packages/odyssey-react-mui/src/MenuItem.tsx

```diff
@@ -10,40 +10,63 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { MenuItem as MuiMenuItem } from "@mui/material";
+import {
+  MenuItem as MuiMenuItem,
+  MenuItemProps as MuiMenuItemProps,
+} from "@mui/material";
 import { menuItemClasses } from "@mui/material/MenuItem";
-import type { MenuItemProps as MuiMenuItemProps } from "@mui/material";
-import { memo, useContext } from "react";
+import {
+  memo,
+  useCallback,
+  useContext,
+  type MouseEventHandler,
+  type ReactNode,
+} from "react";
 
 import { MenuContext } from "./MenuContext";
 
-export interface MenuItemProps
-  extends Omit<
-    MuiMenuItemProps,
-    | "component"
-    | "dense"
-    | "disableGutters"
-    | "divider"
-    | "focusVisibleClassName"
-  > {
-  /**
-   * Toggles whether or not the MenuItem represents a destructive action.
-   */
+export type MenuItemProps = {
+  children: ReactNode;
+  hasInitialFocus?: boolean;
+  isSelected?: boolean;
   isDestructive?: boolean;
-}
+  onClick?: MuiMenuItemProps["onClick"];
+  value?: string;
+  variant?: "default" | "destructive";
+};
 
-const MenuItem = ({ isDestructive, ...props }: MenuItemProps) => {
+const MenuItem = ({
+  children,
+  hasInitialFocus,
+  isSelected,
+  onClick: onClickProp,
+  value,
+  variant = "default",
+}: MenuItemProps) => {
   const { closeMenu } = useContext(MenuContext);
 
+  const onClick = useCallback<MouseEventHandler<HTMLLIElement>>(
```

Comment:
This can actually be changed up to be relative to MUI's prop because that's where it's ultimately going to go:
```suggestion
  const onClick = useCallback<MuiMenuItemProps["onClick"]>(
```

Reply:
<img width="1194" alt="Screenshot 2023-07-07 at 10 53 21 AM" src="https://github.com/okta/odyssey/assets/91091570/ea6fd754-f210-413b-b5c1-a16da94f3545">

It's not happy with me when I do this 🤔

Reply:
`NonNullable<MuiMenuItemProps["onClick"]>`
`<MuiMenuItemProps["onClick"] & {}>`